Source/DiabloUI/settingsmenu.cpp

```diff
@@ -132,8 +142,7 @@ void UiSettingsMenu()
 					vecDialogItems.push_back(std::make_unique<UiListItem>(pCategory->GetName().data(), static_cast<int>(SpecialMenuEntry::None), UiFlags::ColorWhitegold | UiFlags::ElementDisabled));
 					categoryCreated = true;
 				}
-				vecDialogItems.push_back(std::make_unique<UiListItem>(pEntry->GetName().data(), vecOptions.size(), UiFlags::ColorUiGold | UiFlags::NeedsNextElement));
-				vecDialogItems.push_back(std::make_unique<UiListItem>(pEntry->GetValueDescription().data(), vecOptions.size(), UiFlags::ColorUiSilver | UiFlags::ElementDisabled));
+				vecDialogItems.push_back(std::make_unique<UiListItem>("{0}: {1}", CreateDrawStringFormatArgForEntry(pEntry), vecOptions.size(), UiFlags::ColorUiGold));
```

Comment:
This is a very minor comment but there's no need for positional arguments here:

```suggestion
				vecDialogItems.push_back(std::make_unique<UiListItem>("{}: {}", CreateDrawStringFormatArgForEntry(pEntry), vecOptions.size(), UiFlags::ColorUiGold));
```

Reply:
I tried this first but got a crash:

>	devilutionx.exe!std::basic_string_view<char,std::char_traits<char>>::operator[](const unsigned __int64 _Off) Line 1367	C++
 	devilutionx.exe!devilution::`anonymous namespace'::FmtArgParser::operator()(std::basic_string_view<char,std::char_traits<char>> & rest) Line 227	C++
 	devilutionx.exe!devilution::GetLineWidth(std::basic_string_view<char,std::char_traits<char>> fmt, devilution::DrawStringFormatArg * args, unsigned __int64 argsLen, devilution::GameFontTables size, int spacing, int * charactersInLine) Line 382	C++
 	devilutionx.exe!devilution::DrawStringWithColors(const devilution::Surface & out, std::basic_string_view<char,std::char_traits<char>> fmt, devilution::DrawStringFormatArg * args, unsigned __int64 argsLen, const devilution::Rectangle & rect, devilution::UiFlags flags, int spacing, int lineHeight) Line 564	C++
 	devilutionx.exe!devilution::DrawStringWithColors(const devilution::Surface & out, std::basic_string_view<char,std::char_traits<char>> fmt, std::vector<devilution::DrawStringFormatArg,std::allocator<devilution::DrawStringFormatArg>> args, const devilution::Rectangle & rect, devilution::UiFlags flags, int spacing, int lineHeight) Line 194	C++
 	devilutionx.exe!devilution::`anonymous namespace'::Render(const devilution::UiList * uiList) Line 815	C++

>string_view subscript out of range

```cpp
		if (rest.empty() || (rest[1] != '}' && rest.size() < 3)) {
			LogError("Unclosed format argument: {}", fmt_);
		}
		if (rest[2] == '}' && rest[1] >= '0' && rest[1] <= '9') {
```

`rest` has only a size of 2, so `rest[2]` fails.

Reply:
Fixed